doc/http.qbk

```diff
@@ -41,76 +41,236 @@ both Boost.Asio and the HTTP protocol specification described in
+```
+    http::request<http::string_body> req;
+    http::response<http::string_body> resp;
 ```
-http::request<http::empty_body> req({http::method_t::http_get, "/", 11});
-req.headers.insert("Host", "127.0.0.1:80");
-req.headers.insert("User-Agent", "Beast.HTTP");
 
+[heading Members]
+
+Message objects are default constructible, with public access to data members.
+Request and response objects have some common members, and some members unique
+to the message type. These statements set all the members in each message:
+```
+    http::request<http::string_body> req;
+    req.method = http::method_t::http_get;
+    req.url = "/index.html";
+    req.version = 11;           // HTTP/1.1
+    req.headers.insert("User-Agent", "hello_world");
+    req.body = "";
+
+    http::response<http::string_body> resp;
+    resp.status = 404;
+    resp.reason = "Not Found";
+    resp.version = 10;          // HTTP/1.0
+    resp.headers.insert("Server", "Beast.HTTP");
+    resp.body = "The requested resource was not found.";
 ```
 
-When sending a message, the Content-Length and Transfer-Encoding are set
-automatically based on the properties of the body. Depending on the HTTP
-version of the message and properties of the message body, the implementation
-will automaticaly chunk-encode the data sent on the wire:
+The following statements achieve the same effects as the statements above:
 ```
-    ip::tcp::socket sock(ios);
-    ...
-    http::response<http::string_body> resp({200, "OK", 11});
+    http::request<http::string_body> req({http::method_t::http_get, "/index.html", 11});
+    req.headers.insert("User-Agent", "hello_world");
+    req.body = "";
+
+    http::response<http::string_body> resp({404, "Not Found", 10});
     resp.headers.insert("Server", "Beast.HTTP");
-    resp.body = "The command was successful";
+    resp.body = "The requested resource was not found.";
+```
 
-    write(sock, resp);
-}
+[heading Headers]
+
+The `message::headers` member is a container for setting the field/value
+pairs in the message. These statements change the values of the headers
+in the message passed:
 ```
+    template<class Body>
+    void set_fields(http::request<Body>& req)
+    {
+        if(! req.exists("User-Agent"))
+            req.insert("User-Agent", "myWebClient");
 
-Receiving a message is simple, declare a value of type message and call `read`.
-This example reads a message, builds a response, and sends it.
+        if(req.exists("Accept-Charset"))
+            req.erase("Accept-Charset");
+
+        req.replace("Accept", "text/plain");
+    }
 ```
-void handle_connection(ip::tcp::socket& sock)
-{
-    request<string_body> req;
-    read(sock, req);
-    response<string_body> resp;
-    ...
-    write(sock, resp);
-}
+
+[heading Body]
+
+The `message::body` member represents the message body. Depending on the
+`Body` template argument type, this could be a writable container. The library
```

Comment:
I'm reading this as text, so I can't be sure of the formatting.  But I suspect "the library" does not belong at the end of this line.